### PR TITLE
added a new option fix-irreversible-blocks

### PR DESCRIPTION
**Pull request by kimjh2005**
providing --fix-irreversible-blocks flag on eosio-blocklog as in nodeos
## Change Description

eosio-blocklog has a new runtime option: --fix-irreversible-blocks :  When the existing block log is inconsistent with the index, 
allows fixing the block log and index files automatically - that is, it will take the highest indexed block if it is                               valid; otherwise it will repair the block log and reconstruct the index.

The existing --smoke-test flag can be used to test the integrity of the blocks.log and blocks.index files.
## Change Type
**Select ONE**
- [ ] Documentation

- [x] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [x ] Documentation Additions

eosio-blocklog --help will show the following new flag.
--fix-irreversible-blocks             When the existing block log is 
                                        inconsistent with the index, allows 
                                        fixing the block log and index files 
                                        automatically - that is, it will take 
                                        the highest indexed block if it is 
                                        valid; otherwise it will repair the 
                                        block log and reconstruct the index.

### CODE REVIEW

File: libraries/chain/block_log.cpp [hunk 1]
```diff
@@ -700,6 +700,10 @@ namespace eosio { namespace chain {
                         uint16_t max_retained_files, bool fix_irreversible_blocks)
        : my(new detail::block_log_impl(data_dir, archive_dir, stride, max_retained_files, fix_irreversible_blocks)) {}
 
+   block_log::block_log(const fc::path& data_dir,bool fix_irreversible_blocks,
+                        fc::path archive_dir, uint64_t stride, uint16_t max_retained_files)
+       : block_log(data_dir, archive_dir, stride, max_retained_files, fix_irreversible_blocks) {}
+
```
Comment (huangminghuang) [inline libraries/chain/block_log.cpp:706]: Don't create another constructor just to shuffle the input parameters around.

File: libraries/chain/include/eosio/chain/block_log.hpp [hunk 2]
```diff
@@ -36,6 +36,9 @@ namespace eosio { namespace chain {
       public:
          block_log(const fc::path& data_dir, fc::path backup_dir = fc::path(), uint64_t stride=1000, 
                    uint16_t max_retained_files=10, bool fix_irreversible_blocks=false);
+         block_log(const fc::path& data_dir, bool fix_irreversible_blocks,
+                  fc::path backup_dir = fc::path(), uint64_t stride=1000,
+                  uint16_t max_retained_files=10);
```
Comment (huangminghuang) [inline libraries/chain/include/eosio/chain/block_log.hpp:41]: Remove it.

File: programs/eosio-blocklog/main.cpp [hunk 3]
```diff
@@ -233,6 +237,12 @@ bool trim_blocklog_front(bfs::path block_dir, uint32_t n) {        //n is first
    return status;
 }
 
+void fix_irreversible_blocks(bfs::path block_dir) {
+    std::cout << "\nfix_irreversible_blocks of blocks.log and blocks.index in directory " << block_dir << '\n';
+    block_log block_logger(block_dir, true);
```
Comment (huangminghuang) [inline programs/eosio-blocklog/main.cpp:242]: Just use block_log block_logger(block_dir, fc::path(), 0, 0, true);

File: libraries/chain/block_log.cpp (hunk 1 above)
Comment (kimjh2005) [reply libraries/chain/block_log.cpp]: removed it.

File: programs/eosio-blocklog/main.cpp (hunk 3 above)
Comment (kimjh2005) [reply programs/eosio-blocklog/main.cpp]: removed it.

File: libraries/chain/include/eosio/chain/block_log.hpp (hunk 2 above)
Comment (kimjh2005) [reply libraries/chain/include/eosio/chain/block_log.hpp]: removed it.

File: libraries/chain/block_log.cpp [hunk 4]
```diff
@@ -700,7 +700,7 @@ namespace eosio { namespace chain {
                         uint16_t max_retained_files, bool fix_irreversible_blocks)
        : my(new detail::block_log_impl(data_dir, archive_dir, stride, max_retained_files, fix_irreversible_blocks)) {}
 
-   block_log::~block_log() {}
+    block_log::~block_log() {}
```
Comment (huangminghuang) [inline libraries/chain/block_log.cpp:703]: remove the extra space

Comment (kimjh2005) [reply libraries/chain/block_log.cpp]: removed it.

**Pull request merged**